vllm/config/kv_transfer.py

```diff
@@ -79,21 +81,21 @@ def compute_hash(self) -> str:
         hash_str = hashlib.md5(str(factors).encode(), usedforsecurity=False).hexdigest()
         return hash_str
 
-    def __post_init__(self) -> None:
-        if self.engine_id is None:
-            self.engine_id = str(uuid.uuid4())
-
-        if self.kv_role is not None and self.kv_role not in get_args(KVRole):
-            raise ValueError(
-                f"Unsupported kv_role: {self.kv_role}. "
-                f"Supported roles are {get_args(KVRole)}"
-            )
+    @field_validator("engine_id", mode="after")
+    @classmethod
+    def _generate_engine_id(cls, engine_id: Optional[str]) -> str:
+        if engine_id is None:
+            return str(uuid.uuid4())
+        return engine_id
 
+    @model_validator(mode="after")
+    def _validate_kv_role_is_set_if_kv_connector_is_set(self: Self) -> Self:
```

Comment:
Let's just keep the model validator names generic. Specific names won't generalise well if we need to add more validation later.

```suggestion
    def _validate_kv_transfer_config(self: Self) -> Self:
```